packages/next/build/webpack/plugins/flight-client-entry-plugin.ts

```diff
@@ -248,8 +247,56 @@ export class FlightClientEntryPlugin {
       )
     })
 
-    // After optimizing all the modules, we collect the CSS that are still used.
+    // After optimizing all the modules, we collect the CSS that are still used
+    // by the certain chunk.
     compilation.hooks.afterOptimizeModules.tap(PLUGIN_NAME, () => {
+      const cssImportsForChunk: Record<string, string[]> = {}
+
+      function collectModule(entryName: string, mod: any) {
+        const resource = mod.resource
+        if (resource) {
+          if (regexCSS.test(resource)) {
+            cssImportsForChunk[entryName].push(resource)
+          }
+        }
+      }
+
+      compilation.chunkGroups.forEach((chunkGroup: any) => {
```

Comment:
There's a helper `traverseModules` in flight manifest plugin, consider using it to replace the manual chunk/module travsersal?

Reply:
They're not identical though (this one has early exits), but happy to refactor those later!